From 9b320138755542b927df650da0bd1e61ecaa41d7 Mon Sep 17 00:00:00 2001
Message-Id: <9b320138755542b927df650da0bd1e61ecaa41d7.1378117677.git.npajkovs@redhat.com>
From: Vitezslav Samel <vitezslav@samel.cz>
Date: Thu, 29 Aug 2013 10:11:42 +0200
Subject: [PATCH] BUGFIX: fix "Floating point exception" in
 tcplog_flowrate_msg()

commit 0d55bee "tcplog_flowrate_msg(): cleanup and fix") removed
condition, which leads to zero division.

Time diff between current time and ->conn_starttime is 0, because of
rate_print updates happen in less then 1 sec and later on, we try to
divide ->bcount by interval, which is 0, hencs zero division.

Reported-by: Erik K. <ummeegge@ipfire.org>
Signed-off-by: Vitezslav Samel <vitezslav@samel.cz>
Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/tcptable.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tcptable.c b/src/tcptable.c
index a4133d9..e217b19 100644
--- a/src/tcptable.c
+++ b/src/tcptable.c
@@ -437,6 +437,8 @@ static char *tcplog_flowrate_msg(struct tcptableent *entry, char *buf,
 				 size_t bufsize)
 {
 	time_t interval = time(NULL) - entry->conn_starttime;
+	if (interval < 1)
+		interval = 1;
 
 	char rbuf[64];
 	rate_print(entry->bcount / interval, rbuf, sizeof(rbuf));
-- 
1.8.1.2

